checksec.cpp

```diff
@@ -199,8 +199,8 @@ const MitigationReport Checksec::isRFG() const {
     // NOTE(ww): a load config under 92/148 bytes implies the absence of the
     // GuardFlags field. See:
     // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#load-configuration-layout
-    if ((targetMachine_ == peparse::IMAGE_FILE_MACHINE_I386 && loadConfigSize_ < 92) ||
-        loadConfigSize_ < 148) {
+    if (loadConfigSize_ < 148 &&
+        !(imageCharacteristics_ == peparse::IMAGE_FILE_32BIT_MACHINE && loadConfigSize_ >= 92)) {
```

Comment:
I _think_ this check is slightly wrong: `imageCharacteristics_` is a bitfield, with `0x0100` indicating `IMAGE_FILE_32_BIT_MACHINE`.  So, I think this actually needs to be:

```c++
((imageCharacteristics_ & peparse::IMAGE_FILE_32BIT_MACHINE) && loadConfigSize_ >= 92)
```

Otherwise, this'll still be a consistent no-op for the majority of binaries.

Reply:
Whoops, beat me to it!

Reply:
Please squash my sins 😅